src/scales/scale.time.js

```diff
@@ -609,18 +557,17 @@ module.exports = Scale.extend({
 		var timeOpts = options.time;
 		var timestamps = me._timestamps;
 		var ticks = [];
+		var capacity = me.getLabelCapacity(min);
+		var source = options.ticks.source;
+		var distribution = options.distribution;
 		var i, ilen, timestamp;
 
-		switch (options.ticks.source) {
-		case 'data':
+		if (source === 'data' || (source === 'auto' && distribution === 'series')) {
```

Comment:
Why does the auto behaviour change here?

Reply:
Previously when `source` was `auto` we would try to generate ticks that were spaced such that we ended up with a number of labels that would fit alongside each other. Simon had suggested quite awhile ago that he didn't like that we had both this `auto` functionality and the `autoSkip` because they did much of the same thing. So now when `source` is `auto` we just do a dumb generation of ticks at every data point or unit and then apply the `autoSkip`

When `distribution` is `series` then we already have a tick at every data point by simply returning `timestamps.data`, which is very performant and simple. In the case that `distribution` is `linear` we generate a tick every 1 time unit apart because the data points aren't necessarily evenly spaced apart like they are in the `series` case by definition. Besides the fact that it costs more, calling `generate` wouldn't always work in the `series` case. (For an example of where it wouldn't work, consider a `series` chart that plots only weekdays)